### ISSUE TITLE
Empty Future response for successful DeploymentsClient.CreateOrUpdate

### ISSUE THREAD

**Issue by sptramer** [Azure/azure-sdk-for-go#1462]
When running the Go quickstart for VMs (https://docs.microsoft.com/en-us/go/azure/azure-sdk-go-qs-vm) against the latest SDK, the quickstart crashes because it makes the assumption that results returned from deployments will contain data. The offenders appear to be:

```go
	err = deploymentFuture.Future.WaitForCompletion(ctx, deploymentsClient.BaseClient.Client)
	if err != nil {
		log.Fatalf("Error while waiting for deployment creation: %v", err)
	}
	return deploymentFuture.Result(deploymentsClient)
```

```go
	result, err := createDeployment()
	if err != nil {
		log.Fatalf("Failed to deploy correctly: %v", err)
	}
	log.Printf("Completed deployment: %v", *result.Name)
```

The deployment has completed, according to a CLI query:

```json
{
  "id": "/subscriptions/.../resourceGroups/GoVMQuickstart/providers/Microsoft.Resources/deployments/VMDeployQuickstart",
  "name": "VMDeployQuickstart",
...
    "provisioningState": "Succeeded",
```

This crash can be avoided by removing the `Printf` statement here, but this is a __breaking change__ from a release in mid-February and indicates that the SDK may not be properly retrieving information from successful operations. It is unclear if this affects other responses retrieved from futures.

**Comment by jhendrixMSFT** [comment]
I think this is a regression introduced in the fix for https://github.com/Azure/azure-sdk-for-go/issues/1000

**Comment by yangl900** [comment]
This seems an issue for all clients, container instance has the same problem.

I see the referenced issue had this question

>  What is the correct was to determine whether or not we need to perform a final GET to retrieve the result of an async operation? Originally we predicated this based on the polling method, i.e. if the method is Location (returns Location header) then the final response will contain the value, else if it's AsyncOperation (returns Azure-AsyncOperation header) then perform a GET. In this case the 202 returns both Location and Azure-AsyncOperation headers so the code assumes the polling method is AsyncOperation and attempts to perform a GET which is incorrect.

AFAIK Azure-AsyncOperation uri will never return the provisioned resource, regardless it is from a 202 response or 201 response. To get the resource properties, I think we should preform GET on the original resource uri itself.

**Comment by jhendrixMSFT** [comment]
This has been fixed in v17.0.0 of the SDK, https://github.com/Azure/azure-sdk-for-go/releases/tag/v17.0.0
